test/distributed/org/apache/cassandra/distributed/test/topology/TopologyMixupTest.java

```diff
@@ -0,0 +1,856 @@
+                        NodeToolResult result = cluster.get(1).nodetoolResult("cms", "reconfigure", Integer.toString(TARGET_RF));
+                        result.asserts().success();
+                        logger.info("CMS reconfigure: {}", result.getStdout());
+                        preActions.remove(this);
+                    }
+                }
+            });
+            preActions.add(() -> {
+                int[] up = th.up();
+                // use the most recent node just in case the cluster isn't in-sync
+                IInvokableInstance node = cluster.get(up[up.length - 1]);
+                cmsGroup = HackSerialization.cmsGroup(node);
+                currentEpoch.set(HackSerialization.tcmEpoch(node));
+            });
+            List<TransactionalMode> modes = Stream.of(TransactionalMode.values()).filter(t -> t.accordIsEnabled).collect(Collectors.toList());
+            TransactionalMode mode = rs.pick(modes);
+            boolean enableMigration = allowsMigration(mode) && rs.nextBoolean();
+            metadata = fromQT(new CassandraGenerators.TableMetadataBuilder()
+                              .withKeyspaceName(KEYSPACE)
+                              .withTableKinds(TableMetadata.Kind.REGULAR)
+                              .withKnownMemtables()
+                              //TODO (coverage): include "fast_path = 'keyspace'" override
+                              .withTransactionalMode(enableMigration ? TransactionalMode.off : mode)
+                              .withoutEmpty()
+                              .build())
+                       .next(rs);
+            maybeCreateUDTs(cluster, metadata);
+            String schemaCQL = metadata.toCqlString(false, false);
+            logger.info("Creating test table:\n{}", schemaCQL);
+            cluster.schemaChange(schemaCQL);
+            if (enableMigration)
+            {
+                cluster.schemaChange("ALTER TABLE " + metadata + " WITH " + mode.asCqlParam());
+                cluster.get(1).nodetoolResult("consensus_admin", "begin-migration", "--target-protocol", "accord", metadata.keyspace, metadata.name).asserts().success();
+            }
+
+            //TODO (usability): schemaChange does not mean the cluster KNOWS about the table, so replicas may be stale by the time we query
+            // Accord also learns about it async, so every node may know the epoch, but accord doesn't yet!
+            long waitForEpoch = HackSerialization.tcmEpoch(cluster.get(1));
+            for (int i = 1; i <= cluster.size(); i++)
+            {
+                int finalI = i;
+                Awaitility.await()
+                          .atMost(Duration.ofMinutes(1))
+                          .until(() -> HackSerialization.accordEpoch(cluster.get(finalI)) == waitForEpoch);
+            }
+
+            currentEpoch.set(waitForEpoch);
+
+            Gen<Statement> select = (Gen<Statement>) (Gen<?>)  fromQT(new Select.GenBuilder(metadata).withLimit1().build());
```

Comment:
We talked in Slack, it's mostly the lack of coverage.  Right now `AbstractTypeGenerator` supports every single type  (even super weird ones we hope users never touch, but are not blocked...); so we benefit from border coverage.

The other thing is that the new `ast` was built with Accord in mind (I wrote it to help validate Caleb's first CQL integration path), so it coverts more than harry does.  We spoke in slack today about migrating Harry to use the ASTs rather than `StringBuilder` which will make it much easier for harry to do those things as well (examples missing: type hints, `+=` (super common with CAS and txn), etc.).

Now, one thing I did have in mind when writing this test was to make the Accord part pluggable... I just never got around to doing that...  My thinking was that the schema + statements are accord specific but the rest wasn't, so we could make this test abstract and have one for non-accord and one for accord...  the non-accord one would be a good place to use Harry.  Might even be good to have 4 versions of the test: 2 that validate (using harry) and 2 just fuzzing to see what breaks...

Now, we both agree that long term we should converge, which makes these differences disappear...

Reply:
updating more for history than to inform Alex...

Alex and I spoke and worked to get Harry working with Accord... `TopologyMixupTest` got split into 2 tests: with accord / without accord...  the one without accord uses harry for the validation and schema creation.

The accord one still uses this logic as it helps validate the custom CQL that exists for Accord